fix: handle array image edge cases

Prithpal-Sooriya opened the pull request:
## **Description**

Ensures that we handle array images correctly be defaulting to their first image

<!--
Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/30494?quickstart=1)

## **Related issues**

Fixes: https://consensys.slack.com/archives/C8RSKCNCD/p1739984618053419

## **Manual testing steps**

1. Use Flask Build > Enable Watched Accounts
2. Add watched address that owns this NFT: https://polygonscan.com/token/0x4b22c268e3f7f643686e0dfb160b40fe5328cccd?a=0x397d7171504776AD3d5f578ce772914539E28a49 (contract: `0x4b22C268E3f7f643686E0dfB160B40fE5328cccD`, TokenID: `6822`)
3. Select this watched address and navigate to NFT page.
4. Expect NFT to render - (expected image: `ipfs://bafybeidgklvljyifilhtrxzh77brgnhcy6s2wxoxqc2l73zr2nxlwuxfcy`)

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

Prithpal-Sooriya committed:
fix: handle array image edge cases

it ensures that we handle array images correctly be defaulting to their first image

Prithpal-Sooriya committed:
fix: fix more areas that use the image

gambinish commented on ui/components/app/assets/nfts/nft-details/nft-full-image.tsx:
```diff
@@ -81,7 +82,7 @@ export default function NftFullImage() {
                 src={isImageHosted ? image : nftImageURL}
                 alt={nftImageAlt}
-                name={name}
+                name={name ?? undefined}
```
Is this change related to the multiple image fix? Just making sure it's intentional

Prithpal-Sooriya replied:
AH good catch, I was trying to sync all the types... but it was a bit overwhelming to do in 1 PR.
I think this sneaked through 😄

Prithpal-Sooriya replied:
Fixed https://github.com/MetaMask/metamask-extension/pull/30494/commits/5dabaf6f9c13b8b63d7453ddc95a01b0407b21e2

gambinish reviewed (dismissed):
Approving with a small comment

Prithpal-Sooriya committed:
refactor: revert name prop change

This sneaked through while I attempted to synchronise our types. We will need a separate task to do this.

Prithpal-Sooriya committed:
Merge branch 'main' into fix-handle-array-image-edge-case

Prithpal-Sooriya commented on ui/helpers/utils/nfts.ts:
```diff
@@ -28,3 +37,18 @@ export const getNftImageAlt = ({ name, tokenId, description }) => {
+    return image;
+  }
+
+  // TEMP - UI/UX change to support multiple NFT images
+  // or we would need to highly specific logic to better determine which NFT image to view
```
Remove this comment, we have decided not to pursue this, and we will always default to the first image.

Prithpal-Sooriya committed:
Merge branch 'main' of github.com:MetaMask/metamask-extension into fix-handle-array-image-edge-case

Prithpal-Sooriya committed:
chore: remove temp comment

Prithpal-Sooriya committed:
Merge branch 'fix-handle-array-image-edge-case' of github.com:MetaMask/metamask-extension into fix-handle-array-image-edge-case

Prithpal-Sooriya committed:
test: add tests to cover new image logic

Prithpal-Sooriya committed:
Merge branch 'main' of github.com:MetaMask/metamask-extension into fix-handle-array-image-edge-case

Prithpal-Sooriya committed:
Merge branches 'fix-handle-array-image-edge-case' and 'main' of github.com:MetaMask/metamask-extension into fix-handle-array-image-edge-case

OGPoyraz reviewed (approved):
Confirmation changes LGTM
